app/src/full/java/com/celzero/bravedns/scheduler/BugReportZipper.kt

```diff
@@ -228,8 +228,15 @@ object BugReportZipper {
     }
 
     private fun copy(input: InputStream, output: OutputStream) {
-        while (input.read() != -1) {
-            output.write(input.readBytes())
+        try {
+            val buffer = ByteArray(DEFAULT_BUFFER_SIZE)
+            var bytesRead: Int
+            while (input.read(buffer).also { bytesRead = it } != -1) {
+                output.write(buffer, 0, bytesRead)
+            }
+            output.close()
```

Comment:
close in finally?

Also, is there a need for `copy` to close streams when the callers are using `copy` within kt's `use{}` block?

Reply:
This has been removed: https://github.com/celzero/rethink-app/pull/1581/commits/4083e754a30171994dcdf372d1fa12b507860d57